src/org/daisy/dotify/formatter/impl/VolumeProvider.java

```diff
@@ -151,13 +151,18 @@ VolumeImpl nextVolume() {
         List<AnchorData> ad = new ArrayList<>();
         volume.setPreVolData(updateVolumeContents(currentVolumeNumber, ad, true));
         volume.setBody(nextBodyContents(currentVolumeNumber, volume.getOverhead().total(), ad));
-
+        volume.setPostVolData(updateVolumeContents(currentVolumeNumber, ad, false));
         if (logger.isLoggable(Level.FINE)) {
-            logger.fine("Sheets  in volume " + currentVolumeNumber + ": " + (volume.getVolumeSize()) +
-                    ", content:" + volume.getBodySize() +
-                    ", overhead:" + volume.getOverhead());
+            logger.fine("Sheets in volume " + currentVolumeNumber + ": " + (volume.getVolumeSize()) +
+                    " (content:" + volume.getBodySize() +
+                    ", overhead:" + volume.getOverhead().total() + ")");
         }
-        volume.setPostVolData(updateVolumeContents(currentVolumeNumber, ad, false));
+
+        // Calling getSheetsInVolume() before setting the new value because
+        // otherwise the number of sheets in the volume might exceed the maximum
+        // (because the overhead increases) and it will remain unnoticed because
+        // getOverhead() does not change the "dirty" state.
```

Comment:
Would it be better to change `getOverhead()` so that the state becomes dirty?

Reply:
Yes. I was a bit afraid to do it at first, but I've looked into it now and seems safe to do.

Reply:
Done